### PR TITLE
FixOSX Build  fallocate

### PR DESCRIPTION
**Pull request by siying**
The recent update about fallocate failed OSX build. Fix it.

### CODE REVIEW

File: util/io_posix.h [hunk 1]
```diff
@@ -114,9 +114,11 @@ class PosixWritableFile : public WritableFile {
 #ifdef ROCKSDB_FALLOCATE_PRESENT
   virtual Status Allocate(uint64_t offset, uint64_t len) override;
 #endif
+#ifdef OS_LINUX
   virtual Status RangeSync(uint64_t offset, uint64_t nbytes) override;
   virtual size_t GetUniqueId(char* id, size_t max_size) const override;
 };
```
Comment (ajkr) [inline util/io_posix.h]: you need the }; even if non-Linux

Comment (siying) [inline util/io_posix.h]: Thanks! updated.

**Comment by siying** [comment]
Since the build is broken and it takes too long for this diff's turn of Travis. I'm committing it.

**Comment by siying** [comment]
Replaced by #1830

**Pull request closed without merging**